src/AutoRest.CSharp/Common/Output/Models/Types/ModelTypeProvider.cs

```diff
@@ -113,9 +113,27 @@ private ConstructorSignature EnsurePublicConstructorSignature()
 
         private ConstructorSignature EnsureSerializationConstructorSignature()
         {
-            return IncludeDeserializer
-                ? CreateSerializationConstructorSignature(Declaration.Name, Fields.PublicConstructorParameters, Fields.SerializationParameters) ?? InitializationConstructorSignature
-                : InitializationConstructorSignature;
+            // the property bag never needs deserialization, therefore we return the initialization constructor here so that we do not write it in the generated code
+            if (IsPropertyBag)
+                return InitializationConstructorSignature;
+
+            var serializationCtorSignature = CreateSerializationConstructorSignature(Declaration.Name, Fields.PublicConstructorParameters, Fields.SerializationParameters);
+
+            // verifies if this new ctor is the same as the public one
+            if (!serializationCtorSignature.Parameters.Any(p => TypeFactory.IsList(p.Type)) && InitializationConstructorSignature.Parameters.SequenceEqual(serializationCtorSignature.Parameters, new ParameterByTypeEqualityComparer()))
```

Comment:
Do we have any requirements about optional parameters?

Reply:
What do you mean by "requirements"?
This check is here just because we need to ensure we did not write two identical constructors which leads to compile errors.
We will even write something like this:
```
public Foo(IEnumerable<string> bar) {}
internal Foo(IReadOnlyList<string> bar) {}
```
just because they are different. The compiler should be smart enough to figure out which to choose, and mgmt/hlc is always following this way.

For optional parameters, the public constructor will never have them, and serialization constructor will have them therefore they will never be identical constructors.